Tests/FluentAssertions.Specs/Primitives/BooleanAssertionSpecs.cs

```diff
@@ -6,151 +6,200 @@ namespace FluentAssertions.Specs.Primitives;
-    }
-
-    [Fact]
-    public void Should_succeed_when_asserting_boolean_value_to_be_equal_to_the_same_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(false);
-
-        // Assert
-        action.Should().NotThrow();
-    }
-
-    [Fact]
-    public void Should_fail_when_asserting_boolean_value_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(true);
-
-        // Assert
-        action.Should().Throw<XunitException>();
-    }
-
-    [Fact]
-    public void Should_fail_with_descriptive_message_when_asserting_boolean_value_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(true, "because we want to test the failure {0}", "message");
-
-        // Assert
-        action.Should().Throw<XunitException>()
-            .WithMessage("*Expected*boolean*True*because we want to test the failure message, but found False.*");
-    }
-
-    [Fact]
-    public void Should_succeed_when_asserting_boolean_value_not_to_be_equal_to_the_same_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(false);
-
-        // Assert
-        action.Should().NotThrow();
-    }
-
-    [Fact]
-    public void Should_fail_when_asserting_boolean_value_not_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(true);
-
-        // Assert
-        action.Should().Throw<XunitException>();
-    }
-
-    [Fact]
-    public void Should_fail_with_descriptive_message_when_asserting_boolean_value_not_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(true, "because we want to test the failure {0}", "message");
-
-        // Assert
-        action.Should().Throw<XunitException>()
-            .WithMessage("*Expected*boolean*True*because we want to test the failure message, but found True.*");
-    }
-
-    [Fact]
-    public void Should_throw_a_helpful_error_when_accidentally_using_equals()
-    {
-        // Act
-        Action action = () => true.Should().Equals(true);
-
-        // Assert
-        action.Should().Throw<NotSupportedException>().WithMessage("Equals is not part of Fluent Assertions. Did you mean Be() instead?");
+        public static TheoryData<bool?, bool> PassingImplications => new()
+        {
+            { false, false },
+            { false, true },
+            { true, true }
+        };
+
+        public static TheoryData<bool?, bool> NonPassingImplications => new()
+        {
+            { null, true },
+            { null, false },
+            { true, false }
+        };
+
+        [Theory]
+        [MemberData(nameof(PassingImplications), MemberType = typeof(BooleanAssertionSpecs.Imply))]
+        public void Subject_implies_implicator(bool? subject, bool implicator)
```

Comment:
Apparently the left hand side and right hand side of an implication are named "antecedent" and "consequent", respectively

https://en.wikipedia.org/wiki/Material_conditional

Reply:
Ok.. will change that (at least the right hand side)